### PR TITLE
Show assigned user as mention in the task preview

### PR DESCRIPTION
**Pull request by dukenv0307**
### Fixed Issues

$ https://github.com/Expensify/App/issues/19089
PROPOSAL: https://github.com/Expensify/App/issues/19089#issuecomment-1550654651

$ https://github.com/Expensify/App/issues/19959 
$ https://github.com/Expensify/App/issues/19967

### Tests for https://github.com/Expensify/App/issues/19089

1. Login with Expensifail account
2. Go to FAB > Assign task
3. Enter title, description and click on next button
4. Click on assignee  and select any user or enter an email or phone number to select a new user if don't have any user
5. Click on confirm button to create a new task
6. Verify that the assigned user displays as metion in the task preview
7. Edit the title of this task
8. Logout and login again
9. Verify that the assigned users still displays as mention in the task preview as well

### Tests for https://github.com/Expensify/App/issues/19959
1. Go to App and Click on + icon
2. Select Assign task option
3. Create a new task
4. Go to settings > logout
5. Login again
6. Verify that the task name is created in above display

### Tests for https://github.com/Expensify/App/issues/19967
1. Go to any chat
2. Click on + icon and select Assign task option
3. Enter title and description
4. Click on Next button
5. Click on confirm task button
6. click on title
8. Edit task title
10. Go back to previous report screen
11. Verify that Task title is updated

- [x] Verify that no errors appear in the JS console

### Offline tests

Same above
### QA Steps

Same above
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos 
<details>
<summary>Web</summary>

https://github.com/Expensify/App/issues/19089

[Screencast from 02-06-2023 00:47:25.webm](https://github.com/Expensify/App/assets/129500732/72c18e74-208a-44b3-9e1c-f1a09b2710a5)

[Screencast from 06-06-2023 01:44:51.webm](https://github.com/Expensify/App/assets/129500732/aa16ef75-173b-4886-96b6-73a96b85c5cb)

https://github.com/Expensify/App/issues/19959 

https://github.com/Expensify/App/assets/129500732/265bc6e3-b686-4b87-b44a-84fa03271725

https://github.com/Expensify/App/issues/19967

https://github.com/Expensify/App/assets/129500732/e6f36b48-bc0f-484f-bc6c-6b210ddd3070

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/issues/19089

https://github.com/Expensify/App/assets/129500732/080ba31e-5f2b-459c-a1e0-854f50f33962

https://github.com/Expensify/App/assets/129500732/b9947314-d3b8-4e65-85b9-fc6552f9679e

https://github.com/Expensify/App/issues/19959

https://github.com/Expensify/App/assets/129500732/3e25c6aa-45d4-4d9c-a507-3bbd1c82557e

https://github.com/Expensify/App/issues/19967

https://github.com/Expensify/App/assets/129500732/2941006b-d9c5-4244-b501-b110935caae7

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/issues/19089

https://github.com/Expensify/App/assets/129500732/6db85522-8723-4696-9703-d8957367b9b4

https://github.com/Expensify/App/assets/129500732/9660f413-909d-4dc5-a18c-4c671027d9be

https://github.com/Expensify/App/issues/19959 

https://github.com/Expensify/App/assets/129500732/eb31acb8-87d6-40a0-ab39-b378f5d99d95

https://github.com/Expensify/App/issues/19967

https://github.com/Expensify/App/assets/129500732/7ef08616-3a2f-4971-84c6-57eb7afe08fd

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/issues/19089

https://github.com/Expensify/App/assets/129500732/87f8269b-47ac-4f55-885b-762d6e9c250d

https://github.com/Expensify/App/assets/129500732/4b736adb-c309-4865-a082-a8226b4f7401

https://github.com/Expensify/App/issues/19959 

https://github.com/Expensify/App/assets/129500732/7c7f3659-103e-4885-85f8-4448ea14ccf3

https://github.com/Expensify/App/issues/19967

https://github.com/Expensify/App/assets/129500732/3f6fb4f0-c268-45a4-ad0a-155026c9cf3d

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/issues/19089

https://github.com/Expensify/App/assets/129500732/3a831888-faf4-47ff-8fb5-00228bc7af10

https://github.com/Expensify/App/assets/129500732/05155c73-da4c-42a2-b88b-2e5e1ce73b7c

https://github.com/Expensify/App/issues/19959 

https://github.com/Expensify/App/assets/129500732/6be5cd40-b83d-45fe-8d48-d998f4f57a3c

https://github.com/Expensify/App/issues/19967

https://github.com/Expensify/App/assets/129500732/5f49cc47-1bc6-4ed4-b518-8dd251b84b2d

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/issues/19089

https://github.com/Expensify/App/assets/129500732/6ee6fbb5-15c7-4935-98da-405e7a9d09c5

https://github.com/Expensify/App/assets/129500732/0960f550-0630-4231-8b18-561291591256

https://github.com/Expensify/App/issues/19959 

https://github.com/Expensify/App/assets/129500732/e70496b7-ee2e-4ae4-af3e-93fc96ade43d

https://github.com/Expensify/App/issues/19967

https://github.com/Expensify/App/assets/129500732/621d70c2-027f-42d6-a8f9-b99c28ac4df2

</details>

### CODE REVIEW

**Comment by sobitneupane** [comment]
@dukenv0307 Can you please include the test steps from https://github.com/Expensify/App/issues/19959 and https://github.com/Expensify/App/issues/19967 issues and verify that they are no longer reproducible. This PR is expected to solve those issues as well as mentioned by @thienlnam [here](https://github.com/Expensify/App/issues/19089#issuecomment-1572795863).

File: src/components/ReportActionItem/TaskPreview.js [hunk 1]
```diff
@@ -56,8 +56,10 @@ const TaskPreview = (props) => {
     const isTaskCompleted =
         (props.taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.taskReport.statusNum === CONST.REPORT.STATUS.APPROVED) ||
         (props.action.childStateNum === CONST.REPORT.STATE_NUM.SUBMITTED && props.action.childStatusNum === CONST.REPORT.STATUS.APPROVED);
-    const taskTitle = props.action.taskTitle || props.taskReport.reportName;
-    const parentReportID = props.action.parentReportID || props.taskReport.parentReportID;
+    const taskTitle = props.taskReport.reportName || props.action.childReportName;
+    const taskAssignee = props.taskReport.managerEmail || props.action.childManagerEmail || '';
+    const htmlForTaskPreview = taskAssignee === '' ? `<comment>${taskTitle}</comment>` : `<comment><mention-user>@${taskAssignee}</mention-user> ${taskTitle}</comment>`;
```
Comment (thienlnam) [inline src/components/ReportActionItem/TaskPreview.js:61]: htmlForTaskPreview = taskAssignee  ? `<comment><mention-user>@${taskAssignee}</mention-user> ${taskTitle}</comment>` : `<comment>${taskTitle}</comment>`;

**Review by sobitneupane — COMMENTED** [review]
@dukenv0307 When I logout and login again, task assignee goes missing from task preview.


https://github.com/Expensify/App/assets/25876548/eafc3f3e-705b-4afd-b50b-791ebd055f42

**Comment by sobitneupane** [comment]
@dukenv0307

Looks like when I edit the task report title, the task assignee is removed. Might not be related to this PR.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/05ff28ba-7550-42c1-b8a1-ae34625fb9e7

https://github.com/Expensify/App/assets/25876548/6ce44f25-a934-4082-84e6-2a1f370e0e74

Screenshot 2023-06-05 at 13 05 40

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/ee13ead4-2971-4f4c-809e-152d0addb447


</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/d6f2c99a-0d23-4aaf-afc6-01708f37506d

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/25876548/0ca53a54-0bac-454c-aa2e-ae66b650e747

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/b84ab7ff-af95-4627-b5e5-d58b014d4860


</details>

**Comment by sobitneupane** [comment]
Found few issues. Most of them are not related to this PR. Few issues close to this PR:

- @thienlnam Do we want to show taskAssignee in LHN lastMessageText as well?
Screenshot 2023-06-05 at 13 51 00

- @thienlnam Even after the task title is edited, LHN lastMessageText is not updated and shows the old task title. Do we want  to address the issue in this PR?
Screenshot 2023-06-05 at 13 53 30

**Comment by thienlnam** [comment]
> Looks like when I edit the task report title, the task assignee is removed. Might not be related to this PR.

Oh interesting, I'll look into this. Does this happen on main?

> @thienlnam Do we want to show taskAssignee in LHN lastMessageText as well?

No need to show the task assignee in the LHN for now

> @thienlnam Even after the task title is edited, LHN lastMessageText is not updated and shows the old task title. Do we want to address the issue in this PR?

I'm not sure if that can be fixed in the front-end, that probably also requires a BE change

**Review by sobitneupane — DISMISSED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by thienlnam** [comment]
> Yes, It happens in main as well.

Actually, I bet it happens on main because we changed what is returned via the API.
When you edit a task, you always have to provide all information on a task (title, description, assignee) otherwise it will default to null. Can we double check here that when we edit a task we're passing the correct fields to EditTask? I'm guessing this is happening because taskAssignee is null/empty somewhere in the params hence it is getting set to empty

**Review by bondydaa — DISMISSED** [review]
Nothing stands out to me as problematic, 👍

**Comment by bondydaa** [comment]
I'll leave this to @thienlnam to merge once the remaining questions are figured out.

**Comment by thienlnam** [comment]
Yup, great find - could you please make those changes in this PR? This PR handles realigning the use of assignee to managerEmail

**Review by thienlnam — DISMISSED** [review]
Looks great, thanks!

**Review by thienlnam — CHANGES_REQUESTED** [review]
One last thing I noticed as I pulled down your branch, you will have to update these 

    reportAction.reportAction.taskTitle = taskTitle;
    reportAction.reportAction.taskAssignee = taskAssignee;

to use 
childReportName and childManagerEmail

**Review by thienlnam — APPROVED** [review]
Lovely, thank you!

**Pull request merged**